client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -217,6 +394,147 @@ class PaymentMethodStripe extends Component {
 						) }
 					</span>
 				</FormFieldset>
+			</div>
+		);
+	}
+
+	////////////////////////////////////////////////////////////////////////////
+	// Dialog action button methods, including the links that let the user force a flow
+
+	onUserRequestsKeyFlow = () => {
+		this.setState(
```

Comment:
Just curious about the decision to do this in component state rather than redux state...

Reply:
I tend to treat redux state as API sourced (or destined) data and component state as everything else, especially things like child component flow/visibility toggles.

In this particular case, the user can toggle between key flow or connect flow - nothing needs to change in persistent state until they actually connect their account, so I elected to keep that toggle in the containing component's state.  Once they connect an account, the connected account ID (and a few others things) become part of redux state (and are available via API)